lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -74,46 +72,37 @@ def _get(
         """
 
         datasets = params.query.get("datasets")
-        authorized_user: User = Auth.token_auth.current_user()
-
-        if not authorized_user:
-            raise APIAbort(
-                HTTPStatus.UNAUTHORIZED,
-                "User provided access_token is invalid or expired",
-            )
-
-        cache_m = CacheManager(self.config, current_app.logger)
-        benchmark = None
-        benchmark_check = set()
-        stream_file = {}
-
+        benchmark_choice = None
         for dataset in datasets:
-            benchmark = Metadata.getvalue(dataset, "dataset.metalog.pbench.script")
-            benchmark_check.add(benchmark)
-
+            benchmark = Metadata.getvalue(
+                dataset=dataset, key="dataset.metalog.pbench.script"
+            )
             # Validate if all the selected datasets is of same benchmark
-            if len(benchmark_check) != 1:
+            if not benchmark_choice:
+                benchmark_choice = benchmark
+            elif benchmark != benchmark_choice:
                 raise APIAbort(
                     HTTPStatus.BAD_REQUEST,
-                    f"Requested datasets are not of same benchmark. It can't be compared.Benchmarks :{benchmark_check} are requested.",
+                    f"Requested datasets must all use the same benchmark; found references to {benchmark_choice} and {benchmark}.",
```

Comment:
Maybe "Selected dataset benchmarks must match: {benchmark_choice} and {benchmark} cannot be compared."??